### PR TITLE
Join nodes when the selection spans across them

### PR DESCRIPTION
**Pull request by andybalaam**

### CODE REVIEW

File: crates/wysiwyg/src/node_joiner.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            let current_container = current_text.parent_handle();
+
+            // Find the parent that contains both the <b> tags that we are
+            // going to join.
+            let parent = current_container.parent_handle();
+
+            // Look for our sibling container - e.g. the other <b>
+            // TODO: check we have a sibling instead of crashing here
+            let sibling =
+                parent.child_handle(current_container.index_in_parent() + 1);
```
Comment (jmartinesp) [inline crates/wysiwyg/src/node_joiner.rs:97]: While writing the code for formatting several nodes, I added `fn next_sibling() -> DomHandle` and `prev_sibling` for `DomHandle`. Do you think it makes sense to do it there?

File: crates/wysiwyg/src/node_joiner.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+///
+/// For example, if your selection starts in a bold tag, leaves that tag,
+/// and then ends up in another bold tag, the final result should be a
+/// single bold tag containing all your text.
+pub(crate) struct NodeJoiner<S>
+where
+    S: UnicodeString,
+{
+    first_type: Option<String>,
+    last_type: Option<String>,
```
Comment (jmartinesp) [inline crates/wysiwyg/src/node_joiner.rs:31]: For this I would just use `ContainerNodeKind` so we can merge `<b>` and `<strong>`, but it depends on whether we want to do it or not. This approach would also need extra checks for `Formatting` kind, i.e.

**Review by jmartinesp — APPROVED** [review]
Just two comments about possible improvements, but the code LGTM as it is.

File: crates/wysiwyg/src/node_joiner.rs (hunk 1 above)
Comment (andybalaam) [reply crates/wysiwyg/src/node_joiner.rs:97]: Yes, I'll do this next week

File: crates/wysiwyg/src/node_joiner.rs (hunk 2 above)
Comment (andybalaam) [reply crates/wysiwyg/src/node_joiner.rs:31]: Yeah, that sounds good - I'll look next week

**Comment by andybalaam** [comment]
Thanks, I'll merge now to avoid conflicts, and do those improvements on Tuesday.

**Pull request merged**